docs/USAGE-GUIDE.md

```diff
@@ -1777,6 +1777,216 @@ handle. The handle is considered invalid after **s2n_connection_free** is used.
 [s2n_connection_wipe](#s2n\_connection\_wipe) does not need to be called prior to this function. **s2n_connection_free** performs its own wipe
 of sensitive data.
 
+## TLS1.3 Pre-Shared Key Related Calls
+
+S2N-TLS provides support for pre-shared keys (PSKs). PSKs can be established out of band (external PSKs) or in a previous connection (resumption PSKs) for versions >= TLS1.3.
+The PSK is used to mutually authenticate the peers, under the assumption that it is not shared with others.
+This allows the handshake to skip the certificate verification steps, saving bandwidth and latency.
+
+### Benefits of Using Pre-Shared Keys
+
+- Using pre-shared keys can avoid the need for public key operations.
+  This is useful in performance-constrained environments with limited CPU power.
+- PSKs may be more convenient from a key management point of view. For instance, in closed environments where you manually configure the connections in advance, it is easier to configure a PSK than to use certificates. Another case is when the parties already have a mechanism for setting up a shared secret key, and that mechanism could be used to "bootstrap" a key for authenticating the TLS connection.
```

Comment:
```suggestion
- PSKs may be more convenient from a key management point of view. For instance, in closed environments where you manually configure the connections in advance, it is easier to configure a PSK than to use certificates. Another case where it may be more convenient to use PSKs is when the parties already have a mechanism for setting up a shared secret key. That mechanism could be used to "bootstrap" a key for authenticating the TLS connection.
```